src/openrct2-ui/input/ShortcutInput.cpp

```diff
@@ -0,0 +1,407 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2021 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "ShortcutManager.h"
+
+#include <SDL.h>
+#include <cstring>
+#include <openrct2/core/String.hpp>
+
+using namespace OpenRCT2::Ui;
+
+constexpr uint32_t UsefulModifiers = KMOD_SHIFT | KMOD_CTRL | KMOD_ALT | KMOD_GUI;
+
+static uint32_t ParseModifier(const std::string_view& text)
+{
+    if (String::Equals(text, "CTRL", true))
+    {
+        return KMOD_CTRL;
+    }
+    else if (String::Equals(text, "LCTRL", true))
+    {
+        return KMOD_LCTRL;
+    }
+    else if (String::Equals(text, "RCTRL", true))
+    {
+        return KMOD_RCTRL;
+    }
+    else if (String::Equals(text, "SHIFT", true))
+    {
+        return KMOD_SHIFT;
+    }
+    else if (String::Equals(text, "LSHIFT", true))
+    {
+        return KMOD_LSHIFT;
+    }
+    else if (String::Equals(text, "RSHIFT", true))
+    {
+        return KMOD_RSHIFT;
+    }
+    else if (String::Equals(text, "ALT", true))
+    {
+        return KMOD_ALT;
+    }
+    else if (String::Equals(text, "LALT", true))
+    {
+        return KMOD_LALT;
+    }
+    else if (String::Equals(text, "RALT", true))
+    {
+        return KMOD_RALT;
+    }
+    else if (String::Equals(text, "GUI", true))
+    {
+        return KMOD_GUI;
+    }
+    else if (String::Equals(text, "LCTRL", true))
+    {
+        return KMOD_LGUI;
+    }
+    else if (String::Equals(text, "RGUI", true))
+    {
+        return KMOD_RGUI;
+    }
+    else
+    {
+        return 0;
+    }
+}
+
+static uint32_t ParseKey(const std::string_view& text)
+{
+    char buffer[128]{};
+    std::strncpy(buffer, text.data(), sizeof(buffer) - 1);
+    auto keyCode = SDL_GetKeyFromName(buffer);
+    if (keyCode != SDLK_UNKNOWN)
+    {
+        return keyCode;
+    }
+    return 0;
+}
+
+static size_t FindPlus(std::string_view s, size_t index)
+{
+    while (true)
```

Comment:
I think you can use find_last_of instead.

Reply:
This function is to find the next plus (excluding ones we want to ignore) from index, not the last plus.